src/udev/udev-varlink.c

```diff
@@ -0,0 +1,166 @@
+
+static int vl_method_reload(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        assert(link);
+
+        if (sd_json_variant_elements(parameters) > 0)
+                return sd_varlink_error_invalid_parameter(link, parameters);
+
+        log_debug("Received io.systemd.Service.Reload");
+        manager_reload(userdata, /* force = */ true);
+        return sd_varlink_reply(link, NULL);
+}
+
+static int vl_method_set_log_level(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        int r, level;
+
+        assert(link);
+
+        r = varlink_dispatch_set_log_level(link, parameters, &level);
+        if (r < 0)
+                return r;
+
+        log_debug("Received io.systemd.Service.SetLogLevel(%i)", level);
+        manager_set_log_level(userdata, level);
+        return sd_varlink_reply(link, NULL);
+}
+
+static int vl_method_set_children_max(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        unsigned n;
+        int r;
+
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "number", _SD_JSON_VARIANT_TYPE_INVALID, sd_json_dispatch_uint, 0, SD_JSON_MANDATORY },
+                {}
+        };
+
+        assert(link);
+
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, &n);
+        if (r != 0)
+                return r;
+
+        log_debug("Received io.systemd.Udev.SetChildrenMax(%u)", n);
+        manager_set_log_level(userdata, n);
+        return sd_varlink_reply(link, NULL);
+}
+
+static int vl_method_set_environment(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata) {
+        int r;
+
+        static const sd_json_dispatch_field dispatch_table[] = {
+                { "assignments", SD_JSON_VARIANT_ARRAY, sd_json_dispatch_strv, 0, SD_JSON_MANDATORY },
+                {}
+        };
+
+        assert(link);
+
+        _cleanup_strv_free_ char **v = NULL;
+        r = sd_varlink_dispatch(link, parameters, dispatch_table, &v);
+        if (r != 0)
+                return r;
+
+        STRV_FOREACH(s, v)
+                if (!env_assignment_is_valid(*s))
+                        return sd_varlink_error_invalid_parameter(link, parameters);
+
+        log_debug("Received io.systemd.Udev.SetEnvironment()");
+        manager_set_environment(userdata, v);
+        return sd_varlink_reply(link, NULL);
+}
+
+static int vl_method_start_stop_exec_queue(sd_varlink *link, sd_json_variant *parameters, sd_varlink_method_flags_t flags, void *userdata, bool start) {
+        Manager *manager = ASSERT_PTR(userdata);
+        assert(link);
+
+        if (sd_json_variant_elements(parameters) > 0)
+                return sd_varlink_error_invalid_parameter(link, parameters);
+
+        log_debug("Received %s", start ? "io.systemd.Udev.StartExecQueue()" : "io.systemd.Udev.StopExecQueue()");
```

Comment:
we might want to add sd_varlink_get_current_method() that works a bit like sd_varlink_get_current_parameters() which we could use here.